src/main/java/com/mael/ttt/GameSetup.java

```diff
@@ -20,27 +21,44 @@ public GameSetup(Board board, UserInterface gameUI) {
 
     public void playGame() {
         do {
-            gameUI.printWelcomeMessage();
-            board.reset();
-            String option = new Menu(gameUI).getUserOption();
-            Game game     = new Game(new Turn(board, checker, gameUI), getPlayer(option), getOpponent(option));
+            setUp();
+            String option = getMenuOption();
+            Game game     = new Game(new Turn(board, checker, gameUI), createPlayer(option), createOpponent(option));
             game.start();
         } while (gameUI.replay().equals("y"));
     }
 
-    private Player getPlayer(String option) {
-        return MenuOption.createPlayer(getPlayerType(option), gameUI, PLAYER);
+    private void setUp() {
+        board.reset();
+        gameUI.printWelcomeMessage();
+    }
+
+    private String getMenuOption() {
+        return new Menu(gameUI).getUserOption();
+    }
+
+    private Player createPlayer(String option) {
+        return conversionTable(getPlayerType(option), PLAYER);
+    }
+
+    private Player createOpponent(String option) {
+        return conversionTable(getOpponentType(option), OPPONENT);
     }
 
-    private Player getOpponent(String option) {
-        return MenuOption.createPlayer(getOpponentType(option), gameUI, OPPONENT);
+    private PlayerType getPlayerType(String option) {
+        return idToOption(option).getPlayerType();
     }
 
-    private String getPlayerType(String option) {
-        return MenuOption.idToOption(option).getPlayerType();
+    private PlayerType getOpponentType(String option) {
+        return idToOption(option).getOpponentType();
     }
 
-    private String getOpponentType(String option) {
-        return MenuOption.idToOption(option).getOpponentType();
+    public Player conversionTable(PlayerType playerType, Mark mark) {
```

Comment:
What does `conversionTable` mean?

Using a method name like `conversionTable` is like calling it `foo`, `blah` or (_my favourite method name_) `banana`; a meaningless name in the context of a `GameSetup` class.

Looking into the body of the method I can see what it does, and it's not being a conversion table per se. The real functionality this class is offering is building (or creating) a `Player` object from a `String`. Names that come in mind there are something in the line of `createPlayerUsing(String playerAsStrinig)` or `buildPlayerFrom(String option)` (_or something similar_).